chore(internal-release): internal-release 0.7.0

sfoster1 opened the pull request:
This is the tracking branch for internal release 0.7.0.

koji committed:
refactor(component): refactoring icon list (#12598)

add clipboard to icon list

koji committed:
feat(component): add color spacing list to storybook - experiment (#12112)

add color/spacing list to storybook

koji committed:
refactor(app): add constants to constants file for analytics (#12450)

add constants to constants file for analytics

andySigler committed:
make retracting after a capacitive probe optional (#12600)

koji committed:
feat(app): add odd typography and update StyledText (#12535)

Add new typography constants for odd and update StyledText to use 3 types of fontWeights, regular, semiBold, Bold.

jerader committed:
fix(app): when exiting from AttachProbe, display correct text (#12609)

closes RQA-722

b-cooper committed:
fix(app): fix ODD protocol setup step text size (#12614)

koji committed:
fix(app): fix modal header padding to align with hi-fi (#12618)

fix odd modal header padding

ewagoner committed:
Removed user-select ui via css [RAUT-413] (#12559)

vegano1 committed:
fix(hardware): fix the update_fws script so we use target string names. (#12615)

andySigler committed:
pick-up/drop tip settings updates from Carlos tests (#12606)

andySigler committed:
feat(hardware-testing): Volumetric script support multi increment test (#12612)

jerader committed:
refactor(app):get pip cal from instruments endpoint (#12605)

closes RLIQ-386

jerader committed:
refactor(app): attach, detach desktop app design QA (#12611)

closes RLIQ-367

koji committed:
fix(app): use truncation for a long protocol name (#12622)

truncate a long protocol name to avoid breaking a robot card layout

ewagoner committed:
feat(app): labware and modules setup hifi designs (#12582)

* Add deck map icon [RCORE-733]

* Apply hifi designs to labware setup page [RCORE-733]

* Code formatting fixes [RCORE-733]

* Fix small button text color [RCORE-734]

* Create shared back button component [RCORE-734]

* Use new back button in setup headers and remove continue button [RCORE-734]

* Create FloatingActionButton component and use it for Map View button [RCORE-734]

* Add connection-status icon [RCORE-734]

* Fix chip component styling [RCORE-734]

* Create new InlineNotification component [RCORE-734]

* Apply hifi styling to module setup page [RCORE-734]

* Add back info icon [RCORE-734]

* Remove onClick parameter for inline notification [RCORE-734]

* Move ODDBackButton to molecules and add stories and tests [RCORE-734]

* Fix imports, formatting, and general code style improvements [RCORE-734]

* Use new typography [RCORE-734]

* Fix mangled directory names [NOTICKET]

SyntaxColoring committed:
test(api): Configure ctx fixture as PAPIv2.14 when it's configured as an OT-3 (#12567)

ewagoner committed:
Fix race condition when loading pinned protocols [RCORE-752] (#12616)

koji committed:
fix(app): fix UnmatchModuleWarning test (#12628)

fix UnmatchModuleWarning test

ewagoner committed:
feat(app): protocol setup index hifi design (#12626)

* Fix broken UnMatchedModuleWarning test [RCORE-754]

* Fix import of back and more icons to include frame [RCORE-754]

* Add play, pause, and cancel button icons [RCORE-754]

* Apply hifi design [RCORE-754]

* Add todo comment regarding run button components [RCORE-754]

* Fix code prettier issue [RCORE-754]

caila-marashaj committed:
feat(api): allow volume input for blowout (#12534)

TamarZanzouri committed:
feature(api): Allow magnetic block load module in PE (#12613)

jerader committed:
fix(protocol-designer): remove flex tipracks and 96-channel options (#12620)

koji committed:
refactor(app): re-order all text by key's alphabetical order (#12632)

re-order all text by key's alphabetical order

koji committed:
feat(app): add appMode and robotType as new Mixpanel properties (#12519)

add appMode and robotType as new Mixpanel properties

SyntaxColoring committed:
test(robot-server): Wait for server readiness before running tests (#12638)

jbleon95 committed:
fix(api, engine): refactor heater-shaker latch status to prevent false positive move labware check (#12624)

koji committed:
refactor(app): change color name style from snake case to pascalCase with number (#12637)

change color name style from snake case to pascalCase with a number

ewagoner committed:
Alphabetize icon file [RAUT-269] (#12639)

koji committed:
fix(app): Fix Run Completion and RunSummary malfunction (#12625)

Fix Run Completion, Run Summary, and ConfirmRunCancelModal malfunction

jerader committed:
refactor(app): audit of hardcoded slot names, OT-2 and Opentrons Flex (#12635)

koji committed:
chore(app): remove atoms/buttons/OnDeviceDisplay folder and move buttons into atoms/buttons (#12643)

remove atoms/buttons/OnDeviceDisplay folder and move buttons into atoms/buttons

ryanthecoder committed:
feat(hardware) add rear panel testing capability (#12631)

* add messages for aux device changes from the rear panel

* add a method to get all the rear panel pins

* add message to request estop button detection

* spelling fix

* add estop_button_detection to the helper script

koji committed:
feat(app): add network change to robot settings (#12573)

add wifi connection details screen and NetworkDetailsModal

koji committed:
chore(app): remove OnDeviceDisplay folder from atoms/Toast (#12646)

remove OnDeviceDisplay folder from atoms/Toast and slide up stories and test

smb2268 committed:
feat(app): add homing to pipette flows where needed  (#12644)

fix RLIQ-388

jbleon95 committed:
feat(api): return machine appropriate deck slot names in PAPIv2 (#12595)

`ModuleContext.parent` and `Labware.parent` return deck coordinate names

jerader committed:
fix(components): robotWorkSpace remove classnames add transform (#12649)

closes RQA-746

SyntaxColoring committed:
docs(api): Remove accidental public mentions of private members of opentrons.types (#12647)

jerader committed:
feat(app): on pipette flow completion, user is directed to correct page (#12607)

closes RLIQ-387

Co-authored-by: Brian Cooper <brian@opentrons.com>

sfoster1 committed:
chore(internal-release): changes from 0.6.0

koji committed:
feat(app): add clone run to recent run protocol card (#12547)

add re-run function to recent run protocol card with cloning a run

sfoster1 committed:
chore(api): calibrate in a circle (#12653)

Love a circle.

sfoster1 committed:
refactor(api): remove homing when moving to maintain position (#12655)

This reverts commit 337947e242a634efeb83dbaab0441298de9d9ce5.

That commit was itself a revert of 31a8ea8e312de461389a26eca091c2e904f04306 because that commit was merged too early.

The net effect of this commit is to get back to the functionality introduced in #12576

ryanthecoder committed:
fix(hardware): reset eeprom script restarts the fw (#12619)

* turn the app off and on after a reset to make run the update_data_rev task

* use bootloader_for function

shlokamin committed:
refactor(app, api-client, react-api-client): add and use wifi list hook (#12634)

Co-authored-by: Brent Hagen <brent.hagen@opentrons.com>

smb2268 committed:
feat(app, shared-data): wire up pipette flows to protocol setup on app and odd (#12651)

fix RLIQ-217, RLIQ-306, RLIQ-307

TamarZanzouri committed:
feature(api): Allow loading a Magnetic block as a module in PAPI (#12627)

koji committed:
feat(app, app-shell, app-shell-odd): add a new config for initial setup workflow (#12610)

add unfinishedUnboxingFlowRoute, a new config item for unboxing flow

sanni-t commented on api/release-notes-internal.md:
```diff
@@ -41,6 +58,8 @@ Some things are known not to work, and are listed below. Specific compatibility
 - Protocol result screens! No more do you have to stare at those big buttons forever
 
 ### Robot Control
+- Python API `load_labware`, `load_module` and `move_labware` can accept location as deck coordinates (e.g. "A1", "D3") in addition to slot numbers
```
Will need to consolidate these updated notes and the ones listed in 0.7.0 above into THE 0.7.0 notes

TamarZanzouri reviewed (approved):
changes related to mag block looks good!

b-cooper committed:
fix(app): apply current offsets to maintenance run created on LPC launch (#12658)

Copy current offsets over from protocol run into the maintenance run created for each attempt at
LPC.

Closes RQA-755

smb2268 committed:
fix(app): check for Opentrons Flex robot model in useIsOt3 (#12663)

fix RQA-760, RQA-761

koji committed:
fix(app): get back odd menu temp (#12662)

get back odd menu in RobotSettings for network and rename temporarly

TamarZanzouri committed:
fix(api): Allow None serial number when getting module offset (#12665)

koji committed:
feat(app): add rename to robot settings (#12564)

Activate Rename robot in RobotSettings

b-cooper committed:
fix(app): memoize attached pipettes at launch of wizard flow (#12676)

In order to prevent re calculation of the pipette wizard steps, mid-flow, memoize the calculation on
mount

smb2268 committed:
fix(app): fix calibration data source for run setup pipette flows (#12675)

koji committed:
fix(app): fix cancel run freezing issue in ConfirmCancelRunModal (#12660)

fix cancel run freezing issue in ConfirmCancelRunModal

shlokamin committed:
fix(app): check if mixpanel config exists before invoking opt out methods (#12674)

TamarZanzouri committed:
fix(shared-data): Allow mag block in the app (#12684)

b-cooper committed:
fix(app): exclude magnetic block from run setup module matching (#12692)

Because the magnetic block doesn't require a matched attached module in order to complete LPC or a
protocol run, exclude it from the specced to attached module matching process during protocol setup.

Closes RQA-787
